xdl_rs/src/xdl_vec/vec.rs

```diff
@@ -22,14 +22,10 @@ impl DeserializeType for XdlVec {
         for _ in 0..len {
             elements.push(XdlType::deserialize_type(metadata, reader)?);
         }
-        Ok(XdlType::Vec(
-            XdlVec::new(metadata.clone(), elements).map_err(|_| {
-                io::Error::new(
-                    io::ErrorKind::InvalidData,
-                    "All elements are not the same type",
-                )
-            })?,
-        ))
+        Ok(XdlType::Vec(XdlVec::new_unchecked(
```

Comment:
Why did you get rid of the InvalidData error for this? Is it just not needed anymore? think I missed that one.

Also, if we're not returning an io::Error anymore then does the return type still need to be an io::Result?

Reply:
The reason I think the checked constructor can be omitted is because on line 23, XbfType::deserialize_type is being called passing in the inner metadata. If that function were to ever fail to parse whatever is being sent (ie what it tried to parse isn't the same type as all the other things that were being sent), we just exit the function immediately. Therefore if we ever get the the Ok() on line 25 that means we successfully parsed everything as the intended type, so there's no need for the check for homogeneity to occur.